webapp/src/main/java/com/box/l10n/mojito/okapi/filters/AndroidXMLEncoder.java

```diff
@@ -2,41 +2,56 @@
 
 import net.sf.okapi.common.encoder.EncoderContext;
 import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Configurable;
 
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 /**
  * This overrides the {@link net.sf.okapi.common.encoder.XMLEncoder} for Android
  * strings.
- *
+ * <p>
  * It does not escape supported HTML elements for Android strings unless there
  * are variables within the HTML elements.
  * For example, <b>songs</b> vs. &lt;b>%d songs&lt;/b>
- *
+ * <p>
  * Also it overrides the default quotemode setting so the quotes do not get escaped.
- *
+ * <p>
  * For detailed information, see to Android specification in
  * http://developer.android.com/guide/topics/resources/string-resource.html,
  *
  * @author jyi
  */
+@Configurable
 public class AndroidXMLEncoder extends net.sf.okapi.common.encoder.XMLEncoder {
 
+    /**
+     * logger
+     */
+    static Logger logger = LoggerFactory.getLogger(AndroidXMLEncoder.class);
+
     // trying to match variables between html tags, for example, <b>%d</b>, <i>%1$s</i>, <u>%2$s</u>
-    private static final Pattern ANDROID_VARIABLE_WITHIN_HTML = Pattern.compile("(&lt;[b|i|u]&gt;)((.*?)%(([-0+ #]?)[-0+ #]?)((\\d\\$)?)(([\\d\\*]*)(\\.[\\d\\*]*)?)[dioxXucsfeEgGpn](.*?))+(&lt;/[b|i|u]&gt;)");
```

Comment:
@jeeyi not sure to understand that requirement. Do you remember what it was for?

Reply:
I can't remember it anymore. As long as all unit tests work, we can remove it.

Reply:
`testLocalizeAndroidStringsWithEscapedHTMLAndCDATA` seems to test that case, just not sure to understand in that case why it should behave like. I won't touch it